content/en/docs/concepts/services-networking/network-policies.md

```diff
@@ -312,6 +323,60 @@ The value of the label is the namespace name.
 While NetworkPolicy cannot target a namespace by its name with some object field, you can use the
 standardized label to target a specific namespace.
 
+## Pod lifecycle
+
+A NetworkPolicy is applied to a selected pod, when isolation rules are enforced
+for that pod. At that point pod can never establish an outbound connection or receive
+an inbound connection that is not allowed by the NetworkPolicy. Allow rules will be applied
+eventually after the isolation rules (that includes applying both at the same time).
+That means, that once the NetworkPolicy is applied to a pod, the pod should be able to handle situations,
+when it can't establish a connection that is allowed by an existing NetworkPolicy for some time.
+
+Every NetworkPolicy will be applied to all selected pods eventually, but there is no
+way to tell from the existing Kubernetes API when exactly the network policy is applied to a specific
+pod. The only way to check if the network policy is applied to a pod, is to run a connectivity check.
+
+Network plugins may implement extra `status.conditions` for pod readiness based on the
+status of network policies. These conditions then may be used by 
+[Pod readiness gate](/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate).
```

Comment:
This document is written for end users, whereas this comment seems to be targeted more at network plugin authors? We probably shouldn't be suggesting to end users that they can do this if we don't know that there are already network plugins that implement that...